### PR TITLE
Added enhancements to LitmusHttpClient.java and HttpResponseHandler.java

### PR DESCRIPTION
**Pull request by suyeon-jung-dev**
This PR adds:
- Add a feature to the `LitmusHttpClient` class to handle requestParameters.
- Convert the fields under the "data" fields to objects if the response contains a "data" field.
- Add the `BuildRequestTest.java` for testing.

### CODE REVIEW

**Comment by suyeon-jung-dev** [comment]
@jemlog @seedspirit @inpyu @somln Can you please take a look?

Also I adjust builder pattern for request object as well.

**Comment by suyeon-jung-dev** [comment]
Could you guys PTAL?
@jemlog @seedspirit @inpyu @somln

File: src/main/java/io/litmuschaos/request/ListProjectRequest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        this.limit = builder.limit;
+        this.sortField = builder.sortField;
+        this.createdByMe = builder.createdByMe;
+    }
+
+    public static class Builder {
+        private Integer page;
+        private Integer limit;
+        private String sortField;
+        private Boolean createdByMe;
```
Comment (jemlog) [inline src/main/java/io/litmuschaos/request/ListProjectRequest.java:20]: Hi, @suyeon-jung-dev 

Now, we can create ListProjectRequest object like this,
```java
ListProjectRequest request = new ListProjectRequest.Builder().page(1).limit(1).build();
```
I think creating Builder object by `new` keyword is a little bit unnatural. 

How about add static builder() method that return Builder class?
```java
public static Builder builder(){
   return new Builder();
}
```
Then, we can create ListProjectRequest like this,
```java
ListProjectRequest request = ListProjectRequest.builder().page(10).limit(10).build();
```

**Review by seedspirit — APPROVED** [review]
LGTM 👍

**Review by somln — APPROVED** [review]
LGTM 👍

File: src/main/java/io/litmuschaos/request/ListProjectRequest.java (hunk 1 above)
Comment (suyeon-jung-dev) [reply src/main/java/io/litmuschaos/request/ListProjectRequest.java:24]: That make sense. I fixed it in this commit [40ea086](https://github.com/litmuschaos/litmus-java-sdk/pull/24/commits/40ea0868ab7f04b0e70ec0aaa06aaa7447e84e06).

**Comment by suyeon-jung-dev** [comment]
Hi @namkyu1999 
This PR is ready to merge.

cc. @somln

**Review by namkyu1999 — APPROVED** [review]

**Pull request merged**